lib/NormalModule.js

```diff
@@ -442,9 +441,37 @@ class NormalModule extends Module {
 				return callback();
 			}
 
+			const getLoaders = options => {
+				const module =
+					options && options.constructor === Object ? options.module || {} : {};
+				const rules = module.rules || [];
+				if (!rules.length) {
+					return [];
+				}
+				let arr = [];
+				const getLoader = rule => rule.loader || rule.use || "";
+				rules.forEach(rule => {
+					if (Array.isArray(rule.use)) {
+						rule.use.forEach(r => {
+							arr.push(getLoader(r));
+						});
+					} else {
+						arr.push(getLoader(rule));
+					}
+				});
+				return arr;
+			};
+
 			const handleParseError = e => {
 				const source = this._source.source();
-				const error = new ModuleParseError(this, source, e);
+				const loaders = getLoaders(options);
+				const formattedLoaderError = formatLoaderError(loaders);
```

Comment:
formatting the error message should be the job of the `ModuleParseError`. Move it there.